openmls/src/group/mls_group/staged_commit.rs

```diff
@@ -297,29 +278,21 @@ impl MlsGroup {
                 epoch_secrets: provisional_epoch_secrets,
                 interim_transcript_hash,
                 secret_tree: RefCell::new(secret_tree),
-                original_nodes,
+                staged_diff,
             }),
         })
     }
 
     /// Merges a [StagedCommit] into the group state.
-    pub fn merge_commit(&mut self, staged_commit: StagedCommit) {
+    pub fn merge_commit(&mut self, staged_commit: StagedCommit) -> Result<(), MlsGroupError> {
```

Comment:
What kind of error do we expect here? Is it anything more meaningful than a library error?